src/Common/Core/Impl/OS/Win32Process.cs

```diff
@@ -63,6 +63,18 @@ public class Win32Process {
             }
         }
 
+        private void SetExitState() {
+            lock (this) {
```

Comment:
Never use `lock (this)` - your lock can be taken by code outside of class scope and result in a dead-lock.